### PR TITLE
Avoid unused locals for CScript::IsWitnessProgram

### PR DESCRIPTION
**Pull request by Empact**
For those cases where the version and program are not useful.

### CODE REVIEW

**Comment by practicalswift** [comment]
utACK b9f0169bcce3ca7c9d0646ddda5bf264062be87b

Nice simplification

File: src/script/script.cpp [hunk 1]
```diff
@@ -227,6 +227,14 @@ bool CScript::IsWitnessProgram(int& version, std::vector<unsigned char>& program
         return false;
     }
     if ((size_t)((*this)[1] + 2) == this->size()) {
```
Comment (jimpo) [inline src/script/script.cpp]: Might as well return this directly.

**Review by jimpo — APPROVED** [review]
utACK b9f0169

File: src/script/script.cpp (hunk 1 above)
Comment (Empact) [reply src/script/script.cpp]: I like that style too, but held back for the sake of simplifying review. Here's my choice if I'm to simplify the method:

```cpp
static constexpr bool IsOP_N(const unsigned char op)
{
    return (op == OP_0 || (op >= OP_1 && op <= OP_16));
}

// A witness program is any valid CScript that consists of a 1-byte push opcode
// followed by a data push between 2 and 40 bytes.
bool CScript::IsWitnessProgram() const
{
    return ((this->size() >= 4 && this->size() <= 42) &&
            IsOP_N((*this)[0]) &&
            (size_t)((*this)[1] + 2) == this->size());
}
```

Incidentally, this matches the style of `IsPayToWitnessScriptHash` and `IsPayToScriptHash`, above. Thoughts?

Comment (promag) [reply src/script/script.cpp]: @Empact looks good, I'd ACK that.

**Comment by promag** [comment]
utACK b9f0169.

**Review by promag — COMMENTED** [review]
utACK b9f0169.

**Comment by Empact** [comment]
Rebased and update to take advantage of `IsSmallInteger` from #13194

**Comment by sdaftuar** [comment]
I am generally a concept NACK on changes like this; IMO the review burden for refactoring consensus-critical code typically outweighs the benefits.

**Comment by laanwj** [comment]
> I am generally a concept NACK on changes like this; IMO the review burden for refactoring consensus-critical code typically outweighs the benefits.

Agree, seems to risky, closing.

**Pull request closed without merging**